React Capstone - day 3

khanh-devos opened the pull request:
### Tasks in this activity:

- Building a mobile web application to check a list of Nobelists, using React and Redux. The selected API is [api](https://api.nobelprize.org/2.1/laureates).
- Every page (the main page and the pages for each item) should have a unique route within the SPA.
- Follow the design idea by [Nelson Sakwa on Behance](https://www.behance.net/sakwadesignstudio) (colors, layout, typography).
- There are 2 required pages
    1. Home: page shows a list of Nobelists around the world.
    2. Each Country: show Nobelists inside each city of the selected country.
- Testing is required for all redux functions and react components.
-  Deployed by [GitHub Pages](https://pages.github.com/) and the link added inside the readme file.
-  Loom Introduction video is also required (in the README file).

khanh-devos committed:
create react app, config linters: eslint, stylint

khanh-devos committed:
create a new store, use asyncThunk to fetch an API

khanh-devos committed:
Merge pull request #1 from khanh-devos/configReact

React Capstone - day 1

khanh-devos committed:
Set up route based on /country_name and /city_name)

khanh-devos committed:
fix eslint error

khanh-devos committed:
Merge pull request #2 from khanh-devos/buildReduxStore

React Capstone - day 1 # part 2

khanh-devos committed:
Merge pull request #3 from khanh-devos/buildRoutes

React Capstone - day 2

khanh-devos committed:
testing all the components

khanh-devos committed:
fix linter error

khanh-devos committed:
Merge pull request #4 from khanh-devos/buildRoutes

Testing all the components

khanh-devos committed:
Do the styling

khanh-devos committed:
fix linter errors

khanh-devos committed:
Merge pull request #5 from khanh-devos/styling

Do the styling

khanh-devos committed:
deploy by github-pages

khanh-devos committed:
fix linter errors

khanh-devos committed:
Merge pull request #6 from khanh-devos/update

Deployment

khanh-devos committed:
fix basement for all routers

khanh-devos committed:
update meta tags and thumbnail

khanh-devos committed:
update author meta tag

khanh-devos committed:
Merge pull request #7 from khanh-devos/update

Fix basement for all routers

khanh-devos committed:
modify the README file

khanh-devos committed:
Merge pull request #8 from khanh-devos/update

modify the README file

khanh-devos committed:
modify testing

khanh-devos committed:
add Loading state

khanh-devos committed:
add babel config

khanh-devos committed:
Merge pull request #9 from khanh-devos/update

Update

khanh-devos committed:
Add an introduction video

khanh-devos committed:
Merge pull request #10 from khanh-devos/update

Add an introduction video

V-Blaze commented on README.md:
```diff
@@ -0,0 +1,205 @@
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- ACKNOWLEDGEMENTS -->
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+I would like to thank microverse for this project
+
```
> The [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) requires that you give appropriate credit to the author. Therefore, you must do it in the README of your project.

- You have done great so far :+1: :100: :+1: But there are a few requirements you still need to meet before moving to the next milestone. :+1: :+1: 
- [x] As stated in the project requirement, you are required o give credit to the original creator of the design in the `Acknowledgments` section of your README file. I'll suggest you do so. :+1: :+1:

V-Blaze commented on src/redux/nationSlice/NationSlice.jsx:
```diff
@@ -0,0 +1,87 @@
+            prizeAmount: prize,
+          } = item.nobelPrizes[0];
+
+          // const city = city123.replace(/[^a-z0-9]/gi, '_').toLowerCase();
+          // const country = country123.replace(/[^a-z0-9]/gi, '_').toLowerCase();
+
```
- [x] It s against the best practices to leave unused code as a comment on your code base. I'll suggest you remove this as it makes the codebase look a bit unprofessional. :+1: :smile: :wave:

V-Blaze commented on src/redux/nationSlice/NationSlice.jsx:
```diff
@@ -0,0 +1,87 @@
+export const takeUniqueCountries = (arr) => {
+  const uniqueCountries = [];
+  const compare = [];
+  // console.log('takeUniqueCountries'.toUpperCase());
```
- [x] Just as I suggested above, It is against the best practices to leave unused code as a comment on your code base. I'll suggest you remove this as it makes the codebase look a bit unprofessional. :+1: :smile: :wave:

V-Blaze commented on src/test/App.test.js:
```diff
@@ -0,0 +1,227 @@
+import MyHeader from '../components/Header';
+
+jest.mock('axios');
+
```
- [OPTIONAL] You did great writing `unit and integration tests` for all your major components and functions, but following the best practices, I'll suggest you break down these tests into individual multiple files for each component. it will make the test file less bulky and easy to maintain. :+1: :+1:

V-Blaze reviewed (changes requested):
## Hi @khanh-devos 

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

![giphy](https://user-images.githubusercontent.com/107632780/229314554-3700de13-2c00-46d5-bc77-46ad09af7467.gif)


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Good video presentation ✔
-  Project deployed and live link present 👋 
-  correct Git flow✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

khanh-devos commented:
Hi @V-Blaze ,
Thank you for your quick review.
I will fix it now.

khanh-devos committed:
Fix based on the code reviewer's suggestions:
- Remove all unnecessary comments.
- Modify the Acknowledgments of the README file; thanks to the designer.
- Separate the testing into small testing files corresponding to components.

cynthiainga reviewed (approved):
Hi @khanh-devos, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

![](https://user-images.githubusercontent.com/69011963/151541305-67f4f8e4-c79e-4f9d-a6cf-2c7efc3102c6.gif)

## Highlights

- [ ] All requested changes are implemented ✔️ 
- [ ] No linters errors ✔️ 
- [ ] Correct Gitflow ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

khanh-devos commented:
Hi @cynthiainga ,
Thank you for your quick review.
